ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/api/LedgerReader.scala

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.api
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.DamlLogEntryId
+import com.daml.ledger.participant.state.v1.{Configuration, LedgerId, Offset, TimeModel}
+import com.digitalasset.ledger.api.health.{HealthStatus, Healthy}
+@SuppressWarnings(Array("org.wartremover.warts.ArrayEquals"))
+case class LedgerRecord(offset: Offset, entryId: DamlLogEntryId, envelope: Array[Byte])
+
+trait LedgerReader {
+  def events(offset: Option[Offset]): Source[LedgerRecord, NotUsed]
+
+  def retrieveLedgerId(): LedgerId
+
+  def checkHealth(): HealthStatus = Healthy
```

Comment:
Just `extends ReportsHealth` instead. And probably don't provide a default value.